bootloader/src/pyi_launch.c

```diff
@@ -390,53 +385,48 @@ pyi_launch_run_scripts(ARCHIVE_STATUS *status)
         if (ptoc->typcd == ARCHIVE_ITEM_PYSOURCE) {
             /* Get data out of the archive.  */
             data = pyi_arch_extract(status, ptoc);
-            /* Set the __file__ attribute within the __main__ module,
-             *  for full compatibility with normal execution. */
-            namelen = strnlen(ptoc->name, PATH_MAX);
-            if (namelen >= PATH_MAX-strlen(".py")-1) {
-                FATALERROR("Name exceeds PATH_MAX\n");
-                return -1;
-            }
-
-            strcpy(buf, ptoc->name);
-            strcat(buf, ".py");
-            VS("LOADER: Running %s\n", buf);
-
-            if (is_py2) {
-                __file__ = PI_PyString_FromString(buf);
-            }
-            else {
-                __file__ = PI_PyUnicode_FromString(buf);
-            };
-            PI_PyObject_SetAttrString(__main__, "__file__", __file__);
-            Py_DECREF(__file__);
+            VS("LOADER: Running %s.py\n", ptoc->name);
 
             /* Unmarshall code object */
             code = PI_PyMarshal_ReadObjectFromString((const char *) data, ntohl(ptoc->ulen));
-
             if (!code) {
                 FATALERROR("Failed to unmarshal code object for %s\n", ptoc->name);
                 PI_PyErr_Print();
                 return -1;
             }
-            /* Run it */
-            retval = PI_PyEval_EvalCode(code, main_dict, main_dict);
```

Comment:
Do we really need to remove this function? Seems like each iteration of the loop can be this:

```
/* main_dict_copy = copy(main_dict) */
Py_INCREF(main_dict_copy)
retval = PI_PyEval_EvalCode(code, main_dict_copy, main_dict_copy);
```

Reply:
This would be an idea to try. Thanks.

Reply:
This does not work either :-( Same effect as with my code.